pkg/vm/memory/memory_segment.go

```diff
@@ -4,26 +4,25 @@ import (
 	f "github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
 )
 
-type MemorySegment struct {
-	Cells []f.Element
-}
-
-type MemorySegmentManager struct {
+type MemoryManager struct {
+	Memory Memory
 	// where the program byte code is stored
-	ProgramSegment *MemorySegment
-	// stores the function callstack (fp, ap, local variables)
-	ExecutionSegment *MemorySegment
-	// general purpose, for dynamic allocation
-	UserSegment *MemorySegment
-	// each builtin segments is stored here
-	BuiltinSegments *[]MemorySegment
+	// ProgramSegment *MemorySegment
+	//// stores the function callstack (fp, ap, local variables)
+	// ExecutionSegment *MemorySegment
+	//// general purpose, for dynamic allocation
+	// UserSegment *MemorySegment
+	//// each builtin segments is stored here
+	// BuiltinSegments *[]MemorySegment
 }
 
-func CreateMemorySegmentManager(programBytecode *[]f.Element) MemorySegmentManager {
-	return MemorySegmentManager{
-		ProgramSegment:   &MemorySegment{Cells: *programBytecode},
-		ExecutionSegment: nil,
-		UserSegment:      nil,
-		BuiltinSegments:  nil,
+func CreateMemoryManager(programBytecode *[]f.Element) MemoryManager {
```

Comment:
How about changing the argument's name? I mean, the memory manager shouldn't care about bytecode at all. That's out of its abstraction layer.

Reply:
Memory manager just manages memory, y'know

Reply:
Yep it makes sense